[WIP] Module numbering

Kilo59 opened the pull request:
#16 Adding Module numbers

I tried to come up with a module numbering scheme that could be easily checked for with a test.

```python
"""
module doctstring

MODULE:X-Y
"""
```

`X` is the alphabetical order of the Package/directory.
`Y` is the alphabetical order of the Module within the directory.

I'm treating the `__init__.py` module of the root directoy as `MODULE:0-0`

There's probably better ways to do this but I think whatever the scheme out to be deterministic/testable.

## TODO

- [x] Add test to validate module number in docstings
- [ ] Add module numbers to errors? (or save for another PR)

DahlitzFlorian commented:
@Kilo59 module numbering looks good so far. I guess it would be good practice to create a new PR for module numbering in errors after this one got merged.
You've already mentioned, that tests need to be added. I would add another TODO for you if it's possible.

- [ ] Pycodestyle should check for existing pattern in docstring (MODULE:X-Y)

Kilo59 commented:
@DahlitzFlorian I updated the module numbers to reflect new changes and added a basic test that checks for the presence of a module number. It doesn't check for correctness of the module number yet. I wanted to get some feedback about my approach.

I was planning on changing the test to use [regex](https://pythex.org/?regex=MODULE%3A(%3FP%3Cpkg%3E%5Cd)-(%3FP%3Cmod%3E%5Cd)&test_string=A%20module%20for%20working%20with%20the%20.wily%2F%20cache%20directory.%0A%0AThis%20API%20is%20not%20intended%20to%20be%20public%20and%20should%20not%20be%20consumed%20directly.%0AThe%20API%20in%20this%20module%20is%20for%20archivers%20and%20commands%20to%20work%20with%20the%20local%20cache%0A%0AMODULE%3A0-2&ignorecase=0&multiline=0&dotall=0&verbose=0) `MODULE:(?P<pkg>\d)-(?P<mod>\d)` and then walk though the wily directories to make sure the `pkg` and `mod` numbers are correct.


I didn't have a chance to figure out how I would create a custom `Pycodestyle` to check for the pattern.
